net: ipv4: Allow IGMP packets with zero src address

ad8sweera opened the pull request:
IGMP queries sent out by a proxy querier can have
a source IP address 0.0.0.0
Allow these incoming packets.

Fixes #69917

jukkar commented on subsys/net/ip/ipv4.c:
```diff
@@ -317,7 +317,8 @@ enum net_verdict net_ipv4_input(struct net_pkt *pkt, bool is_loopback)
 	}
```
Please change the commit subject to `net: ipv4: Allow IGMP packets with zero src address`
and also add `Fixes #69917` before the signed-off-by line so that the commit is linked to the issue.

jukkar commented on subsys/net/ip/ipv4.c:
```diff
@@ -317,7 +317,8 @@ enum net_verdict net_ipv4_input(struct net_pkt *pkt, bool is_loopback)
 	if (net_ipv4_is_addr_unspecified((struct in_addr *)hdr->src) &&
-	    !net_ipv4_is_addr_bcast(net_pkt_iface(pkt), (struct in_addr *)hdr->dst)) {
+	    !net_ipv4_is_addr_bcast(net_pkt_iface(pkt), (struct in_addr *)hdr->dst) &&
+		(hdr->proto != IPPROTO_IGMP)) {
```
Could you align the added line to same column as the previous line.

jukkar reviewed (commented):
Looks good, couple of minor nits.

rlubos reviewed (commented):
Please squash the commits

ad8sweera committed:
net: ipv4: Allow IGMP packets with zero src address

IGMP queries sent out by a proxy querier can have
a source IP address 0.0.0.0
Allow these incoming packets.

Fixes #69917

Signed-off-by: Savin Weeraratne <savin.weeraratne@audinate.com>

ad8sweera commented:
Thanks all, will someone do the merge? (I don't have permission)

pdgendt commented:
> Thanks all, will someone do the merge? (I don't have permission)

The merge will be done by administrators, but it needs approval from the assignee first, so please wait for a review.
